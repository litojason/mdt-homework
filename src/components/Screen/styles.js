import styled from "styled-components";

import { colors, spacing } from "constants/styles";

export const Screen = styled.div`
  display: flex;
  flex-direction: column;
  height: 100vh;
  box-sizing: border-box;
  background-color: ${colors.background};
`;
export const Content = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
  padding: ${spacing.base};
`;
export const AuthFooter = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: ${spacing.half};
`;
