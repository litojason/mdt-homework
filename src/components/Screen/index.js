import React from "react";

import { Screen as StyledScreen, Content } from "./styles";
import Navbar from "components/Navbar";

const Screen = ({ hasNavbar, children }) => {
  return (
    <StyledScreen>
      {hasNavbar ? (
        <>
          <Navbar />
          <Content>{children}</Content>
        </>
      ) : (
        <Content>{children}</Content>
      )}
    </StyledScreen>
  );
};

export default Screen;
