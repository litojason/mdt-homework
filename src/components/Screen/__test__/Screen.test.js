import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import Screen from "../index";

describe("Component: Screen", () => {
  it("should render Screen", () => {
    const component = <Screen />;

    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render Screen with Navbar", () => {
    const component = <Screen hasNavbar />;

    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
