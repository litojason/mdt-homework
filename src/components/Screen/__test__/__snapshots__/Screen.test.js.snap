// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Screen should render Screen 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100vh;
  box-sizing: border-box;
  background-color: #F5F5F5;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  padding: 1rem;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;
